MobileWallet/Screens/Send/AddRecipient/AddRecipientViewController.swift

```diff
@@ -435,8 +442,10 @@ class AddRecipientViewController: UIViewController, UITextFieldDelegate, Contact
     }
 
     @objc private func onContinue() {
+        selectedContact = contactsTableVC.getContact(publicKey: selectedRecipientPublicKey)
         let amountVC = AddAmountViewController()
         amountVC.publicKey = selectedRecipientPublicKey
+        amountVC.title = selectedContact?.alias.0
```

Comment:
I think you should propagate the contact rather than setting the title property of the VC - VC should be responsible for how to render the data. @SerhiiShovkoplias

Reply:
Yes, I did this initially, but later I saw that the contact in these controllers was used only for the title. It seemed to me superfluous and I decided to transfer only the title. If you think to propagate Contact property is better i will do it

Reply:
Yeah I think the VCs should have access to the contact, and it should propagate all the way up to the SendingTari VC. So from a VC's perspective it should either have a public key or a contact (which provides public key already).